crates/release_plz/tests/all/release_pr.rs

```diff
@@ -315,6 +314,96 @@ async fn release_plz_doesnt_add_invalid_labels_to_release_pr() {
     }
 }
 
+#[tokio::test]
+#[cfg_attr(not(feature = "docker-tests"), ignore)]
+async fn release_pr_works_with_publish_false() {
+    const CARGO_PUBLISH_FALSE_CRATE: &str = "cargo-publish-false";
+    let context = TestContext::new_workspace(&[CARGO_PUBLISH_FALSE_CRATE, "foo"]).await;
+
+    let config = r#"
+[workspace]
+publish = false
+    "#;
+
+    context.write_release_plz_toml(config);
+    context.run_release_pr().success();
+
+    let cargo_publish_false_crate_dir = context.repo_dir().join(CARGO_PUBLISH_FALSE_CRATE);
+
+    // Write publish = false to Cargo.toml
+    let cargo_toml_path = cargo_publish_false_crate_dir.join("Cargo.toml");
+    let mut cargo_toml = LocalManifest::try_new(&cargo_toml_path).unwrap();
+    cargo_toml.data["package"]["publish"] = false.into();
+    cargo_toml.write().unwrap();
+
+    let expected_title = "chore: release v0.1.0";
+    let opened_prs = context.opened_release_prs().await;
+    assert_eq!(opened_prs.len(), 1);
+    assert_eq!(opened_prs[0].title, expected_title);
+
+    context.merge_release_pr().await;
+    // The commit contains the PR id number
+    let expected_commit = format!("{expected_title} (#1)");
+    assert_eq!(
+        context.repo.current_commit_message().unwrap(),
+        expected_commit
+    );
+
+    context.run_release().success();
```

Comment:
Similar as the other comment: should we test that the package was not published in the registry?